src/main/scala/lenthall/spray/SprayCanHttpService.scala

```diff
@@ -75,8 +75,8 @@ object SprayCanHttpService {
     : Future[Unbound] = {
       unbind(duration) andThen {
         case _ =>
-          actorSystem.shutdown()
-          actorSystem.awaitTermination()
+          actorSystem.terminate()
+          Await.ready(actorSystem.whenTerminated, Duration.Inf)
```

Comment:
Because that's now what akka tells you to do when you want to block until termination. See the docs